src/OhmmsPETE/OhmmsVector.h

```diff
@@ -131,12 +128,12 @@ class Vector
     if (n > nAllocated)
     {
       resize_impl(n);
-      if (MemType == MemorySpace::HOST)
+      if (allocator_traits<Alloc>::is_host_accessible)
```

Comment:
I'm not fond of this return to runtime branching on allocator_traits. But the whole nAllocated, nLocal, n + attacheReference situation is not real clean and probably should be resolved in another PR first.

Reply:
The attachReference is for other purpose and I agree it needs improving.
Here I prefer focusing on this PR and get it finished.
I will move the branch outside.